### PR TITLE
chore(wkt): improved wkt typings

### PR DESCRIPTION
**Pull request by Evin-K**

### CODE REVIEW

File: modules/wkt/src/lib/parse-wkt.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export default function parseWKT(input) {
   function multicoords() {
     white();
     var depth = 0;
-    var rings = [];
+    var rings = Array();
```
Comment (belom88) [inline modules/wkt/src/lib/parse-wkt.ts:50]: What advantage of this way of array initialization?

Comment (belom88) [reply modules/wkt/src/lib/parse-wkt.ts:50]: I suggest replacing all `var` declarations with `const` of `let` int this file.

**Review by belom88 — CHANGES_REQUESTED** [review]
`modules/wkt/src/workers/wkt-worker.js → modules/wkt/src/workers/wkt-worker.ts` - edit scripts in `package.json`

**Comment by ibgreen** [comment]
I just landed https://github.com/visgl/loaders.gl/pull/1705 which touched the WKB parts but not the WKT parts.

Just rebase and focus on the WKT improvements.

**Review by belom88 — APPROVED** [review]

File: modules/wkt/src/lib/encode-wkt.ts [hunk 2]
```diff
@@ -4,28 +4,40 @@
...
+ * @returns string
+ */
+
+export default function encodeWKT(gj: {
+  type: string;
+  properties?: {};
+  geometry?: any;
+  coordinates?: any;
+  geometries?: any;
+}) {
```
Comment (dryabinin-actionengine) [inline modules/wkt/src/lib/encode-wkt.ts:19]: I think the return value can be added right after arguments list:
```
export default function encodeWKT(gj: {
  type: string;
  properties?: {};
  geometry?: any;
  coordinates?: any;
  geometries?: any;
}): string
```

Comment (dryabinin-actionengine) [reply modules/wkt/src/lib/encode-wkt.ts:19]: Also should we avoid `any` type in `encodeWKT` function arguments?
Can we just be more accurate with typings here?

File: modules/wkt/src/lib/encode-wkt.ts [hunk 3]
```diff
@@ -4,28 +4,40 @@
...
   if (gj.type === 'Feature') {
     gj = gj.geometry;
   }
 
-  function pairWKT(c) {
+  function pairWKT(c: any[]) {
```
Comment (dryabinin-actionengine) [inline modules/wkt/src/lib/encode-wkt.ts:24]: Should we detect what particular type here?
Do they always have different types?

**Review by dryabinin-actionengine — CHANGES_REQUESTED** [review]

File: modules/wkt/src/lib/encode-wkt.ts [hunk 4]
```diff
@@ -4,28 +4,40 @@
 /**
  * Stringifies a GeoJSON object into WKT
  */
-export default function encodeWKT(gj) {
+
+/**
+ * @param gj
```
Comment (ibgreen) [inline modules/wkt/src/lib/encode-wkt.ts:9]: Nit: I don't like abbreviations, type out as `geojson`...

**Review by ibgreen — APPROVED** [review]
This looks like a solid improvement. Agree it would  be ideal to replace all occurrences of `any`. This could also be done a second PR.

**Review by dryabinin-actionengine — APPROVED** [review]

File: modules/wkt/src/lib/encode-wkt.ts [hunk 5]
```diff
@@ -4,44 +4,56 @@
...
 
-  function wrapParens(s) {
+  function wrapParens(s: string) {
     return '(' + s + ')';
   }
 
   switch (gj.type) {
     case 'Point':
-      return 'POINT (' + pairWKT(gj.coordinates) + ')';
+      return 'POINT (' + pairWKT(gj.coordinates as any[]) + ')';
```
Comment (ibgreen) [inline modules/wkt/src/lib/encode-wkt.ts:46]: Can we do a follow up PR where we remove this `any`?

**Review by ibgreen — APPROVED** [review]

**Pull request merged**